src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/hosts/host-form/host-form.component.ts

```diff
@@ -57,50 +66,99 @@ export class HostFormComponent extends CdForm implements OnInit {
   }
 
   private createForm() {
     this.hostForm = new CdFormGroup({
       hostname: new FormControl('', {
-        validators: [
-          Validators.required,
-          CdValidators.custom('uniqueName', (hostname: string) => {
-            return this.hostnames && this.hostnames.indexOf(hostname) !== -1;
-          })
-        ]
+        validators: this.hostnameValidators
       }),
       addr: new FormControl('', {
         validators: [CdValidators.ip()]
       }),
       labels: new FormControl([]),
+      pattern: new FormControl(false),
       maintenance: new FormControl({ value: false, disabled: this.pageURL !== 'hosts' })
     });
   }
 
   submit() {
+    const hostnameArray = [];
     const hostname = this.hostForm.get('hostname').value;
+    if (this.hostForm.get('pattern').value === true) {
+      if (hostname.includes('[') && hostname.includes(']')) {
+        const hostnameRange = hostname.replace(/(^.*\[|\].*$)/g, '');
+        const patternBrackets = hostname.substring(
+          hostname.indexOf('['),
+          hostname.indexOf(']') + 1
+        );
+        const indexes = hostnameRange.split('-');
+        let hostNumber: string;
+        for (let i = indexes[0]; i <= indexes[indexes.length - 1]; i++) {
+          if (i > 0 && i < 10) {
+            hostNumber = '0' + i;
+          } else {
+            hostNumber = i;
+          }
+          hostnameArray.push(hostname.replace(patternBrackets, hostNumber));
+        }
+      } else if (hostname.includes(',')) {
+        let hostArray = [];
+        hostArray = hostname.split(',');
+        hostArray.forEach((host: any) => {
+          hostnameArray.push(host);
+        });
+      } else {
+        hostnameArray.push(hostname);
+      }
+    } else {
+      hostnameArray.push(hostname);
+    }
     this.addr = this.hostForm.get('addr').value;
     this.status = this.hostForm.get('maintenance').value ? 'maintenance' : '';
     this.allLabels = this.hostForm.get('labels').value;
     if (this.pageURL !== 'hosts' && !this.allLabels.includes('_no_schedule')) {
       this.allLabels.push('_no_schedule');
     }
-    this.taskWrapper
-      .wrapTaskAroundCall({
-        task: new FinishedTask('host/' + URLVerbs.ADD, {
-          hostname: hostname
-        }),
-        call: this.hostService.create(hostname, this.addr, this.allLabels, this.status)
-      })
-      .subscribe({
-        error: () => {
-          this.hostForm.setErrors({ cdSubmitButton: true });
-        },
-        complete: () => {
-          this.pageURL === 'hosts'
-            ? this.router.navigate([this.pageURL, { outlets: { modal: null } }])
-            : this.activeModal.close();
-        }
-      });
+    hostnameArray.forEach((hostnames) => {
+      this.taskWrapper
+        .wrapTaskAroundCall({
+          task: new FinishedTask('host/' + URLVerbs.ADD, {
+            hostname: hostnames
+          }),
+          call: this.hostService.create(hostnames, this.addr, this.allLabels, this.status)
```

Comment:
So it means for 100 hostnames list 100 API request!! ?🙄 Should we make some changes in backend to instead take list of hostnames and forward request synchronously to cephadm `orch host add` ? @alfonsomthd @epuertat

Reply:
Also it'll take "years" to get inventory, facts, etc as it'll refresh `@forallhosts()` in cephadm `serve.py` for 10/100s of hosts.

Reply:
> So it means for 100 hostnames list 100 API requests!! ? Should we make some changes in backend to instead take list of hostnames and forward request synchronously to cephadm `orch host add` ? @alfonsomthd @epuertat

I think so. Maybe we should create a `bulk` option/endpoint to handle this at the backend level. Does
orchestrator support bulk host addition?

Reply:
> Does orchestrator support bulk host addition?

I think not.